### ISSUE TITLE
installation torch-scatter with torch 1.10.1+cu113

### ISSUE THREAD

**Issue by Jdear17** [rusty1s/pytorch_scatter#467]
I'm trying to install PointNeXt, in the install.sh it suggest using: 1.10.1+cu113, However since my problem is just creating an environment with torch-scatter I have asked the question here, hope that is OK. on your page there is no mention of cuda 11.3 just 11.8 and newer. I'm really at a loss and have been stuck for a long time thank you.

When I install torch-scatter on the environment it fails when i run: 

pip install torch-scatter -f https://data.pyg.org/whl/torch-1.10.1+cu113.html

gives: 
(openpoints) bash-4.4$ pip install --user torch-scatter -f https://data.pyg.org/whl/torch-1.10.1+cu113.html
Looking in links: https://data.pyg.org/whl/torch-1.10.1+cu113.html
Collecting torch-scatter
  Using cached torch_scatter-2.1.1.tar.gz (107 kB)
  Preparing metadata (setup.py) ... done
Building wheels for collected packages: torch-scatter
  Building wheel for torch-scatter (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [34 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.linux-x86_64-3.7
      creating build/lib.linux-x86_64-3.7/torch_scatter
      copying torch_scatter/__init__.py -> build/lib.linux-x86_64-3.7/torch_scatter
      copying torch_scatter/placeholder.py -> build/lib.linux-x86_64-3.7/torch_scatter
      copying torch_scatter/scatter.py -> build/lib.linux-x86_64-3.7/torch_scatter
      copying torch_scatter/segment_coo.py -> build/lib.linux-x86_64-3.7/torch_scatter
      copying torch_scatter/segment_csr.py -> build/lib.linux-x86_64-3.7/torch_scatter
      copying torch_scatter/testing.py -> build/lib.linux-x86_64-3.7/torch_scatter
      copying torch_scatter/utils.py -> build/lib.linux-x86_64-3.7/torch_scatter
      creating build/lib.linux-x86_64-3.7/torch_scatter/composite
      copying torch_scatter/composite/__init__.py -> build/lib.linux-x86_64-3.7/torch_scatter/composite
      copying torch_scatter/composite/logsumexp.py -> build/lib.linux-x86_64-3.7/torch_scatter/composite
      copying torch_scatter/composite/softmax.py -> build/lib.linux-x86_64-3.7/torch_scatter/composite
      copying torch_scatter/composite/std.py -> build/lib.linux-x86_64-3.7/torch_scatter/composite
      running egg_info
      warning: no previously-included files matching '*' found under directory 'test'
      writing manifest file 'torch_scatter.egg-info/SOURCES.txt'
      running build_ext
      creating build/temp.linux-x86_64-3.7
      creating build/temp.linux-x86_64-3.7/csrc
      creating build/temp.linux-x86_64-3.7/csrc/cpu
      gcc -pthread -B /rds/general/user/jpd17/home/.conda/envs/openpoints/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -DWITH_PYTHON -Icsrc -I/rds/general/user/jpd17/home/.local/lib/python3.7/site-packages/torch/include -I/rds/general/user/jpd17/home/.local/lib/python3.7/site-packages/torch/include/torch/csrc/api/include -I/rds/general/user/jpd17/home/.local/lib/python3.7/site-packages/torch/include/TH -I/rds/general/user/jpd17/home/.local/lib/python3.7/site-packages/torch/include/THC -I/rds/general/user/jpd17/home/.conda/envs/openpoints/include/python3.7m -c csrc/cpu/scatter_cpu.cpp -o build/temp.linux-x86_64-3.7/csrc/cpu/scatter_cpu.o -O3 -Wno-sign-compare -DAT_PARALLEL_OPENMP -fopenmp -DTORCH_API_INCLUDE_EXTENSION_H -DPYBIND11_COMPILER_TYPE=\"_gcc\" -DPYBIND11_STDLIB=\"_libstdcpp\" -DPYBIND11_BUILD_ABI=\"_cxxabi1011\" -DTORCH_EXTENSION_NAME=_scatter_cpu -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++14
      cc1plus: warning: command line option '-Wstrict-prototypes' is valid for C/ObjC but not for C++
      gcc -pthread -B /rds/general/user/jpd17/home/.conda/envs/openpoints/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -DWITH_PYTHON -Icsrc -I/rds/general/user/jpd17/home/.local/lib/python3.7/site-packages/torch/include -I/rds/general/user/jpd17/home/.local/lib/python3.7/site-packages/torch/include/torch/csrc/api/include -I/rds/general/user/jpd17/home/.local/lib/python3.7/site-packages/torch/include/TH -I/rds/general/user/jpd17/home/.local/lib/python3.7/site-packages/torch/include/THC -I/rds/general/user/jpd17/home/.conda/envs/openpoints/include/python3.7m -c csrc/scatter.cpp -o build/temp.linux-x86_64-3.7/csrc/scatter.o -O3 -Wno-sign-compare -DAT_PARALLEL_OPENMP -fopenmp -DTORCH_API_INCLUDE_EXTENSION_H -DPYBIND11_COMPILER_TYPE=\"_gcc\" -DPYBIND11_STDLIB=\"_libstdcpp\" -DPYBIND11_BUILD_ABI=\"_cxxabi1011\" -DTORCH_EXTENSION_NAME=_scatter_cpu -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++14
      cc1plus: warning: command line option '-Wstrict-prototypes' is valid for C/ObjC but not for C++
      g++ -pthread -shared -B /rds/general/user/jpd17/home/.conda/envs/openpoints/compiler_compat -L/rds/general/user/jpd17/home/.conda/envs/openpoints/lib -Wl,-rpath=/rds/general/user/jpd17/home/.conda/envs/openpoints/lib -Wl,--no-as-needed -Wl,--sysroot=/ build/temp.linux-x86_64-3.7/csrc/cpu/scatter_cpu.o build/temp.linux-x86_64-3.7/csrc/scatter.o -L/rds/general/user/jpd17/home/.local/lib/python3.7/site-packages/torch/lib -lc10-ltorch -ltorch_cpu -ltorch_python -o build/lib.linux-x86_64-3.7/torch_scatter/_scatter_cpu.so -s
      /rds/general/user/jpd17/home/.conda/envs/openpoints/compiler_compat/ld: cannot find /lib/libpthread.so.0: No such file or directory
      /rds/general/user/jpd17/home/.conda/envs/openpoints/compiler_compat/ld: cannot find /usr/lib/libpthread_nonshared.a: No such file or directory
      collect2: error: ld returned 1 exit status
      error: command '/apps/jupyterhub/latest/g++' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for torch-scatter
  Running setup.py clean for torch-scatter
Failed to build torch-scatter
Installing collected packages: torch-scatter
  Running setup.py install for torch-scatter ... error
  error: subprocess-exited-with-error

  × Running setup.py install for torch-scatter did not run successfully.
  │ exit code: 1
  ╰─> [36 lines of output]
      running install
      /rds/general/user/jpd17/home/.local/lib/python3.7/site-packages/setuptools/command/install.py:37: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        setuptools.SetuptoolsDeprecationWarning,
      running build
      running build_py
      creating build
      creating build/lib.linux-x86_64-3.7
      creating build/lib.linux-x86_64-3.7/torch_scatter
      copying torch_scatter/__init__.py -> build/lib.linux-x86_64-3.7/torch_scatter
      copying torch_scatter/placeholder.py -> build/lib.linux-x86_64-3.7/torch_scatter
      copying torch_scatter/scatter.py -> build/lib.linux-x86_64-3.7/torch_scatter
      copying torch_scatter/segment_coo.py -> build/lib.linux-x86_64-3.7/torch_scatter
      copying torch_scatter/segment_csr.py -> build/lib.linux-x86_64-3.7/torch_scatter
      copying torch_scatter/testing.py -> build/lib.linux-x86_64-3.7/torch_scatter
      copying torch_scatter/utils.py -> build/lib.linux-x86_64-3.7/torch_scatter
      creating build/lib.linux-x86_64-3.7/torch_scatter/composite
      copying torch_scatter/composite/__init__.py -> build/lib.linux-x86_64-3.7/torch_scatter/composite
      copying torch_scatter/composite/logsumexp.py -> build/lib.linux-x86_64-3.7/torch_scatter/composite
      copying torch_scatter/composite/softmax.py -> build/lib.linux-x86_64-3.7/torch_scatter/composite
      copying torch_scatter/composite/std.py -> build/lib.linux-x86_64-3.7/torch_scatter/composite
      running egg_info
      warning: no previously-included files matching '*' found under directory 'test'
      writing manifest file 'torch_scatter.egg-info/SOURCES.txt'
      running build_ext
      creating build/temp.linux-x86_64-3.7
      creating build/temp.linux-x86_64-3.7/csrc
      creating build/temp.linux-x86_64-3.7/csrc/cpu
      gcc -pthread -B /rds/general/user/jpd17/home/.conda/envs/openpoints/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -DWITH_PYTHON -Icsrc -I/rds/general/user/jpd17/home/.local/lib/python3.7/site-packages/torch/include -I/rds/general/user/jpd17/home/.local/lib/python3.7/site-packages/torch/include/torch/csrc/api/include -I/rds/general/user/jpd17/home/.local/lib/python3.7/site-packages/torch/include/TH -I/rds/general/user/jpd17/home/.local/lib/python3.7/site-packages/torch/include/THC -I/rds/general/user/jpd17/home/.conda/envs/openpoints/include/python3.7m -c csrc/cpu/scatter_cpu.cpp -o build/temp.linux-x86_64-3.7/csrc/cpu/scatter_cpu.o -O3 -Wno-sign-compare -DAT_PARALLEL_OPENMP -fopenmp -DTORCH_API_INCLUDE_EXTENSION_H -DPYBIND11_COMPILER_TYPE=\"_gcc\" -DPYBIND11_STDLIB=\"_libstdcpp\" -DPYBIND11_BUILD_ABI=\"_cxxabi1011\" -DTORCH_EXTENSION_NAME=_scatter_cpu -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++14
      cc1plus: warning: command line option '-Wstrict-prototypes' is valid for C/ObjC but not for C++
      gcc -pthread -B /rds/general/user/jpd17/home/.conda/envs/openpoints/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -DWITH_PYTHON -Icsrc -I/rds/general/user/jpd17/home/.local/lib/python3.7/site-packages/torch/include -I/rds/general/user/jpd17/home/.local/lib/python3.7/site-packages/torch/include/torch/csrc/api/include -I/rds/general/user/jpd17/home/.local/lib/python3.7/site-packages/torch/include/TH -I/rds/general/user/jpd17/home/.local/lib/python3.7/site-packages/torch/include/THC -I/rds/general/user/jpd17/home/.conda/envs/openpoints/include/python3.7m -c csrc/scatter.cpp -o build/temp.linux-x86_64-3.7/csrc/scatter.o -O3 -Wno-sign-compare -DAT_PARALLEL_OPENMP -fopenmp -DTORCH_API_INCLUDE_EXTENSION_H -DPYBIND11_COMPILER_TYPE=\"_gcc\" -DPYBIND11_STDLIB=\"_libstdcpp\" -DPYBIND11_BUILD_ABI=\"_cxxabi1011\" -DTORCH_EXTENSION_NAME=_scatter_cpu -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++14
      cc1plus: warning: command line option '-Wstrict-prototypes' is valid for C/ObjC but not for C++
      g++ -pthread -shared -B /rds/general/user/jpd17/home/.conda/envs/openpoints/compiler_compat -L/rds/general/user/jpd17/home/.conda/envs/openpoints/lib -Wl,-rpath=/rds/general/user/jpd17/home/.conda/envs/openpoints/lib -Wl,--no-as-needed -Wl,--sysroot=/ build/temp.linux-x86_64-3.7/csrc/cpu/scatter_cpu.o build/temp.linux-x86_64-3.7/csrc/scatter.o -L/rds/general/user/jpd17/home/.local/lib/python3.7/site-packages/torch/lib -lc10-ltorch -ltorch_cpu -ltorch_python -o build/lib.linux-x86_64-3.7/torch_scatter/_scatter_cpu.so -s
      /rds/general/user/jpd17/home/.conda/envs/openpoints/compiler_compat/ld: cannot find /lib/libpthread.so.0: No such file or directory
      /rds/general/user/jpd17/home/.conda/envs/openpoints/compiler_compat/ld: cannot find /usr/lib/libpthread_nonshared.a: No such file or directory
      collect2: error: ld returned 1 exit status
      error: command '/apps/jupyterhub/latest/g++' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> torch-scatter

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
